lib/screens/v2/widget_instance/dup_alert.ex

```diff
@@ -94,16 +94,14 @@ defmodule Screens.V2.WidgetInstance.DupAlert do
 
     parameters = {effect, location, affected_line_count, t.primary_section_count, rotation_index}
 
-    delay_severity = t.alert.severity
+    special_case = alert_layout_special_case(t, parameters)
```

Comment:
I think one thing that makes this a little confusing is that rotation 2 is considered a special case. I don't really think of it as special in the same sense that delays are. It just uses a different set of rules because it (potentially) holds secondary info. (Hopefully that makes sense) Is it possible to handle primary and secondary rotation layouts in different functions? So you could pattern match on `t.rotation_index` and pull all of the logic starting at line 226 into the function for rotation index 2. With that separation, the table still holds up because it is used exclusively by the primary rotations. What do you think?

Reply:
Thanks for nudging me in the right direction! Hopefully the updated approach makes more sense. It separates delay alerts out right at the top now, and the table only gives the mapping for rotation `zero`.

Reply:
The changes are substantial enough that it will probably be easier to review the diff in the split view now.